custom/ewsghana/reports/email_reports.py

```diff
@@ -63,12 +66,14 @@ def _stock_status(status, loc):
                 case_id=location.supply_point_id,
                 section_id=STOCK_SECTION_TYPE,
                 sql_product__in=location_products
-            )
+            ).select_related('sql_product')
 
             for product in location_products:
                 row_data[product.name]['total_fac'] += 1
 
             for state in stock_states:
+                if state.sql_product not in location_products:
```

Comment:
isn't this a duplication of logic since you already have `sql_product__in=location_products` in the query?